components/threading/global_threads/thread_list/virtualized_thread_list.tsx

```diff
@@ -0,0 +1,91 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {memo, useRef, useCallback, useMemo} from 'react';
+import AutoSizer from 'react-virtualized-auto-sizer';
+import {DynamicSizeList} from 'dynamic-virtualized-list';
+
+import {$ID} from 'mattermost-redux/types/utilities';
+import {UserThread} from 'mattermost-redux/types/threads';
+
+import ThreadItem from '../thread_item';
+
+type Props = {
+    ids: Array<$ID<UserThread>>;
+    selectedThreadId?: $ID<UserThread>;
+};
+
+const ITEMS_ON_SCREEN = 8;
+const OVERSCAN_COUNT_BACKWARD = 4;
+const OVERSCAN_COUNT_FORWARD = 4;
+
+const virtListStyles = {
+    maxHeight: '100%',
+    position: 'absolute',
+    top: '0',
+};
+
+function VirtualizedThreadList({
+    ids,
+    selectedThreadId,
+}: Props) {
+    const innerRef = useRef();
+
+    const renderRow = ({itemId}: {itemId: $ID<UserThread>}) => {
+        return (
+            <ThreadItem
+                key={itemId}
+                threadId={itemId}
+                isSelected={selectedThreadId === itemId}
+            />
+        );
+    };
+
+    const canLoadMorePosts = useCallback(() => {}, []);
```

Comment:
Doesn't this need to return something? I would've expected something with this name to return true/false.

Reply:
@hmhealey well I'd expect the same, but seeing the source code for `dynamic-sized-list` it seems that it's more of a callback and not a guard on something: [L519](https://github.com/mattermost/dynamic-virtualized-list/blob/master/src/DynamicSizeList.js#L519)

But this made realize a bug... the list doesn't get updated automatically on new threads getting added, let me fix that.